airbyte-integrations/connectors/source-bing-ads/source_bing_ads/source.py

```diff
@@ -0,0 +1,309 @@
+            )
+
+    def get_account_id(self, stream_slice: Mapping[str, Any] = None) -> Optional[str]:
+        """
+        Fetches account_id from slice object
+        """
+        return stream_slice.get("account_id") if stream_slice else None
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        stream_state = stream_state or {}
+        next_page_token = None
+
+        while True:
+            params = self.request_params(
+                stream_state=stream_state,
+                stream_slice=stream_slice,
+                next_page_token=next_page_token,
+            )
+
+            response = self.send_request(params, account_id=self.get_account_id(stream_slice))
+            for record in self.parse_response(response):
+                yield record
+
+            next_page_token = self.next_page_token(response, current_page_token=next_page_token)
+            if not next_page_token:
+                break
+
+        yield from []
+
+
+class Accounts(BingAdsStream):
+    """
+    Searches for accounts that the current authenticated user can access.
+    API doc: https://docs.microsoft.com/en-us/advertising/customer-management-service/searchaccounts?view=bingads-13
+    Account schema: https://docs.microsoft.com/en-us/advertising/customer-management-service/advertiseraccount?view=bingads-13
+    """
+
+    data_field: str = "AdvertiserAccount"
+    service_name: str = "CustomerManagementService"
+    operation_name: str = "SearchAccounts"
+    additional_fields: str = "TaxCertificate AccountMode"
+    # maximum page size
+    limit: int = 1000
+
+    def next_page_token(self, response: sudsobject.Object, current_page_token: Optional[int]) -> Optional[Mapping[str, Any]]:
+        current_page_token = current_page_token or 0
+        if response is not None and hasattr(response, self.data_field):
+            return None if self.limit > len(response[self.data_field]) else current_page_token + 1
+        else:
+            return None
+
+    def request_params(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+        predicates = {
+            "Predicate": [
+                {
+                    "Field": "UserId",
+                    "Operator": "Equals",
+                    "Value": self.config["user_id"],
+                }
+            ]
+        }
+
+        if self.config["accounts"]["type"] == "subset":
+            predicates["Predicate"].append(
+                {
+                    "Field": "AccountId",
+                    "Operator": "In",
+                    "Value": ",".join(self.config["accounts"]["ids"]),
```

Comment:
and this line could be modified like this:
`"Value": ",".join(self.config["account_ids"]),`

Reply:
After these changes, the customer leaves the `account_ids` field in the UI config as empty, by default, which means - "all" accounts would be fetched. 
If the list of account_ids is provided in the UI, we fetch only them.